tests/deploy_test.go

```diff
@@ -13,6 +14,75 @@ import (
 
 var emptyQueryString = ""
 
+func Test_Scaling_Up(t *testing.T) {
+	functionRequest := requests.CreateFunctionRequest{
+		Image:      "functions/alpine:latest",
+		Service:    "test-scaling",
+		Network:    "func_functions",
+		EnvProcess: "env",
+	}
+
+	deployStatus, deployErr := deploy(t, functionRequest)
+	if deployErr != nil {
+		t.Errorf(deployErr.Error())
+		t.Fail()
+		return
+	}
+
+	if deployStatus != http.StatusOK && deployStatus != http.StatusAccepted {
+		t.Errorf("got %d, wanted %d or %d", deployStatus, http.StatusOK, http.StatusAccepted)
+	}
+
+	invoke(t, functionRequest.Service, emptyQueryString, http.StatusOK)
+
+	scaleStatus, err := scale(functionRequest.Service, "2")
+	if err != nil {
+		t.Error(err)
+	}
+	if scaleStatus != http.StatusOK && scaleStatus != http.StatusAccepted {
+		t.Errorf("got %d, wanted %d or %d", scaleStatus, http.StatusOK, http.StatusAccepted)
+	}
+
+	replicas, getReplicasStatus, err := getReplicas(functionRequest.Service)
+	if err != nil {
+		t.Error(err)
+	}
+	if getReplicasStatus != http.StatusOK && getReplicasStatus != http.StatusAccepted {
+		t.Errorf("got %d, wanted %d or %d", getReplicasStatus, http.StatusOK, http.StatusAccepted)
+	}
+
+	want := "2"
+	if want != fmt.Sprintf("%d", replicas) {
+		t.Errorf("want %s replicas, Got %d", want, replicas)
+	}
+}
+
+func scale(fnName string, replicas string) (int, error) {
+	request := `{"service": "` + fnName + `", "replicas": ` + replicas + `}`
```

Comment:
We have a struct that can be used for this, maybe you can copy/paste it here rather than vendoring.

String concatenation for JSON is not very maintainable in the long-run. Can you change it please?

Reply:
https://github.com/openfaas/faas-provider/blob/8f7c35975e1b2bf8286c2f90ee51633eec427491/types/requests.go#L6

Reply:
I'm in two minds between copying it and vendoring it. What did we do elsewhere? cc @LucasRoesler

Reply:
@alexellis, I agree that we should not do string concatenation for JSON. I'll change that.

I slightly prefer vendoring it, but I'm ok with copy/paste. I'll check other repositories (like openfaas/faas) to see what we do elsewhere.But before I choose which way to pull in the struct, I'll wait for @LucasRoesler's feedback.

Reply:
Let's vendor, faas-provider was made as an interface, these tests are supposed to be using OpenFaaS like a black box, but I think that this is OK.